### PR TITLE
Add text color unittest

### LINKED ISSUE #614: add test cases to compare generated png and expected png.
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

In reference to the discussion: https://github.com/fossasia/badgeyay/pull/615

### PR DESCRIPTION
**Pull request by rupav**
Fixes #614

#### Checklist

- [x ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Added unit test for testing if adding `txt_color` value while generating badges results in successful response.
- File changed : `test_api.py`

### CODE REVIEW

**Comment by rupav** [comment]
@abishekvashok @vaibhavsingh97

File: backend/app/tests/test_api.py [hunk 1]
```diff
@@ -71,6 +71,19 @@ def test_custom_font(self):
...
+        # Tests using a color as background
+        File = {'file': open(self.CSVPath, 'rb')}
+        Data = {
+            'img-default': 'user_defined.png',
+            'bg_color': '000000'
+            'txt_color': '000000'
+        }
+        sleep(2)
+        r = requests.post(self.URL, data=Data, files=File, timeout=10)
+        self.assertEqual(json.loads(r.text)['response'], self.Responses['test_txt_color'])
```
[the remark this answers is not in the document]
Comment (abishekvashok) [reply backend/app/tests/test_api.py]: @rupav consider making this short

[the remark this answers is not in the document]
Comment (rupav) [reply backend/app/tests/test_api.py]: I tried with `/` but hound would still shout. I will give it a try again with a different approach.

File: backend/app/tests/test_api.py [hunk 2]
```diff
@@ -71,6 +71,20 @@ def test_custom_font(self):
...
+        File = {'file': open(self.CSVPath, 'rb')}
+        Data = {
+            'img-default': 'user_defined.png',
+            'bg_color': '000000',
+            'txt_color': '000000'
+        }
+        sleep(2)
+        r = requests.post(self.URL, data=Data, files=File, timeout=10)
+        self.assertEqual(json.loads(r.text)['response'],
+          self.Responses['test_txt_color'])
```
[the remark this answers is not in the document]
Comment (abishekvashok) [reply backend/app/tests/test_api.py]: @rupav

**Comment by rupav** [comment]
It's done now @abishekvashok

**Comment by rupav** [comment]
Actually these indeed are vague, but you can refer similar tests. Like custom font test, background color test.

**Comment by rupav** [comment]
But this test is left out.

**Comment by gabru-md** [comment]
Improving the quality of tests is something we should work upon, and not just adding `vague tests`, which you also have the same point of view upon, I guess.

Let me add @ParthS007 to this discussion. His assistance would be much needed here i suppose. :)

**Comment by abishekvashok** [comment]
This can be done by comparing against a set of badges generated correctly I guess.

**Comment by harshit98** [comment]
Hey there 👋 @rupav. It seems that you're not following up the best practices: https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia/

Since you're adding unit tests, you should have removed the **preview link** section from PR details. The changes which you're describing in the PR detail is not clear.

**Comment by rupav** [comment]
@harshit98 updated my PR description and removed PR link.
This test added is similar to other unittests in `test_api.py` file, but also checks for `txt_color` parameter while generating badges.

**Comment by rupav** [comment]
@abishekvashok my approach would be: Adding tests which check for validation with generated badges' details, i.e. through LXML parsing. But for that, we need to have `png_to_svg` (which is impractical) and then check for badges details. I think here we can't do much with this approach.
Can you please explain your approach?

**Comment by abishekvashok** [comment]
Compare the generated png with a proper png @rupav

**Comment by rupav** [comment]
So comparing 2 .pngs, we need to use other libraries, like https://pypi.python.org/pypi/dfrnt/0.2, but don't you think that would be an overkill? 
`Another approach`: Instead we can test the functions involved while generating, example testing `do_text_fill` in `svg_to_png`?

**Comment by abishekvashok** [comment]
Nope. It would just generate and compare. Whats the overkill?

**Comment by rupav** [comment]
I am concerned with other dependencies for making these tests, yes we can go ahead with that, comparing 2 pngs, lets wait for others to put their opinions too.

**Comment by vaibhavsingh97** [comment]
@rupav I would agree with @abishekvashok , please compare generated PNG  and Proper PNG. Please update your PR

**Comment by vaibhavsingh97** [comment]
@rupav Yeah, We can do that too. Cool! I am closing this PR. Feel free to open issue and fix the same

**Pull request closed without merging**